### PR TITLE
clh: Enable net device hotplug support

### LINKED ISSUE #453: Add hotplug net device support for cloud hypervisor
**Is your feature request related to a problem? Please describe.**

Add hotplug net device support for cloud hypervisor

**Describe the solution you'd like**

Use cloud hypervisor `/vm.add-net` API to hotplug net device

**Describe alternatives you've considered**

**Additional context**

Currently executing `docker network connect test_container test_net` has no effect, and the log shows the following error
> Jul 24 17:09:46 A23074915-33 kata-runtime[9023]: time="2020-07-24T17:09:46.142298499+08:00" level=error msg="Error attach virtual ep" arch=amd64 command=kata-network container=aec7f1d3a255f9132a2e33326e312549de87eb1c01c8ddd41dcc11da1db859a4 error="cannot hotplug device: unsupported device type '2'" name=kata-runtime pid=9023 sandbox=aec7f1d3a255f9132a2e33326e312549de87eb1c01c8ddd41dcc11da1db859a4 source=virtcontainers subsystem=network

### PR DESCRIPTION
**Pull request by BetaXOi**
This patch enables the net device hotplug with cloud hypervisor.

Fixes #453

Signed-off-by: NingBo <ning.bo9@zte.com.cn>

### CODE REVIEW

**Comment by devimc** [comment]
@likebreath @sboeuf  please take a look

File: src/runtime/virtcontainers/clh.go [hunk 1]
```diff
@@ -455,6 +457,48 @@ func (clh *cloudHypervisor) hotPlugVFIODevice(device config.VFIODev) error {
...
+		}
+	default:
+		return fmt.Errorf("this endpoint(%v) is not supported", endpoint.Type())
+	}
+
+	cl := clh.client()
+	ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+	defer cancel()
+
+	_, _, err := cl.VmmPingGet(ctx)
```
Comment (sboeuf) [inline src/runtime/virtcontainers/clh.go:489]: Please remove this as this is not needed and might slow down the code in production.

**Comment by likebreath** [comment]
@BetaXOi Thanks for the contributions. The PR itself looks good to me.

As you may (not) know, the CI jobs related to cloud-hypervisor (clh) are not yet enabled for Kata 2.0 (this repository). Also, I believe there are quite some patches on clh support from kata-runtime repo that needs to be ported to Kata 2.0. I think we'd better hold on merging the PR until we have those CI jobs running and the patches ported (tracked here https://github.com/kata-containers/kata-containers/issues/461).

File: src/runtime/virtcontainers/clh.go (hunk 1 above)
Comment (Marshalzxy) [reply src/runtime/virtcontainers/clh.go:489]: Why do we need to ping there? @BetaXOi

Comment (likebreath) [reply src/runtime/virtcontainers/clh.go:489]: This is a legacy code path from `clh.go` in general. I will have a PR to remove all  the `VmmPingGet` in Kata 1.x (kata-runtime repo).  We will port that change to kata 2.0 while we enable clh CI here.

@BetaXOi In this PR, you can safely assume that a call to `VmmPingGet` is not required. Thanks.

Comment (BetaXOi) [reply src/runtime/virtcontainers/clh.go]: `VmmPingGet` has been removed. thanks.

**Comment by likebreath** [comment]
@BetaXOi We now have a working CI for testing CLH in kata 2.0 (see PR https://github.com/kata-containers/kata-containers/pull/543 and PR https://github.com/kata-containers/tests/pull/2797). 

Please rebase the PR and we will be able to merge it as long as related CI jobs are passing. Thanks again for patience and contributions.

**Comment by likebreath** [comment]
@BetaXOi Can you please back-port this patch to kata 1.x (kata-runtime)?  Kata 1.x is still being actively used, and it is important to keep both versions up-to-date in terms of functionality. Also, we will have much better CI coverage in 1.x for clh. Thanks again for the contributions.

**Comment by BetaXOi** [comment]
Okay, I will back-port it to 1.x first, and then merge it to 2.x at an appropriate time in the future

**Pull request closed without merging**